src/lexer.ls

```diff
@@ -1239,46 +1247,89 @@ character = if not JSON? then uxxxx else ->
             if tokens[i + 2].0 is 'BY'
                 tokens[i + 2].0 = 'RANGE_BY'
             token.op = token.1
-            from-num = 0
+            nFrom = 0
             fallthrough
         case 'RANGE'
+            # {{{
+            # prepare
             lno = token.2
             cno = token.3
-            if from-num? or (tokens[i-1].0 is '['
-            and tokens[i + 1].0 is 'STRNUM'
-            and ((tokens[i + 2].0 is ']'
-                and (tokens[i + 1].1.char-at(0) in ['\'' '"']
-                    or +tokens[i + 1].1 >= 0))
-                or (tokens[i + 2].0 is 'RANGE_BY'
-                and tokens[i + 3]?.0 is 'STRNUM'
-                and tokens[i + 4]?.0 is ']')))
-                unless from-num?
-                    [from-num, char] = decode token.1, lno
-                [to-num, tochar] = decode tokens[i + 1].1, lno
-                carp 'bad "to" in range' lno if not to-num? or char .^. tochar
-                by-num = 1
-                if byp = tokens[i + 2]?.0 is 'RANGE_BY'
-                    carp 'bad "by" in range' tokens[i + 2].2 unless by-num = +tokens[i + 3]?.1
-                else if from-num > to-num
-                    by-num = -1
-                ts = []
-                enc = if char then character else String
-                add = !->
-                     if 0x10000 < ts.push ['STRNUM' enc n; lno, cno] [',' ',' lno, cno]
-                         carp 'range limit exceeded' lno
-                if token.op is 'to'
-                    for n from from-num to to-num by by-num then add!
-                else
-                    for n from from-num til to-num by by-num then add!
-                ts.pop! or carp 'empty range' lno
-                tokens.splice i, 2 + 2 * byp, ...ts
-                i += ts.length - 1
+            # check token sequence,
+            # range with accessor should not be expanded
+            if tokens[i - 2].0 != 'DOT' and
+               # implicit range (when <from> flag-number is set) or,
+               (nFrom != null or
+               # there is a "simple" numeric range startup with..
+               (tokens[i - 1].0 == '[' and
+                tokens[i + 1].0 == 'STRNUM' and
+                # either end of the range..
+                ((tokens[i + 2].0 == ']' and
+                  # with a string literal(?) or a positive number..
+                  # Q: why this exact check order should matter?
+                  (tokens[i + 1].1.0 == '"' or
```

Comment:
`"abc"[0 to 3]` IIRC compiles to something different than `("abc")[0 to 3]`

Reply:
started debugger with `"abc"[0 to 3]` and got these tokens:
```
0: (4) ["NEWLINE", "↵", 0, 0, eol: true, spaced: true]
1: (4) ["STRNUM", ""abc"", 1, 5]
2: (4) ["DOT", ".", 1, 5]
3: (4) ["[", "[", 1, 5]
4: (4) ["RANGE", "0", 1, 6, spaced: true, op: "to"]
5: (4) ["STRNUM", "3", 1, 11]
6: (4) ["]", "]", 1, 12, eol: true, spaced: true, index: true]
7: (4) ["NEWLINE", "↵", 3, 0, spaced: true]
```
at i == 4...  so it checks `'3' === '"'`...